common/buildcraft/lib/net/MessageUpdateTile.java

```diff
@@ -46,6 +49,7 @@ public void fromBytes(ByteBuf buf) {
     public void toBytes(ByteBuf buf) {
         PacketBuffer buffer = new PacketBuffer(buf);
         buffer.writeBlockPos(pos);
+        buffer.writeString(className);
```

Comment:
i know this doesn't fix the core bug, just makes the workaround lighter on the networking